std/file.d

```diff
@@ -539,10 +539,20 @@ version(Posix) private void writeImpl(const(char)[] name, const(FSChar)* namez,
     cenforce(fd != -1, name, namez);
     {
         scope(failure) core.sys.posix.unistd.close(fd);
+
         immutable size = buffer.length;
-        cenforce(
-            core.sys.posix.unistd.write(fd, buffer.ptr, size) == size,
-            name, namez);
+        immutable chks = size_t.max / 2 + 1;
```

Comment:
You are not copying data from one file to another, you are writing data that's *already* in memory to disk. All those links talk about copying data from one file to another. In this case, using a buffer smaller than the biggest one is suboptimal.

`1<<30` because it is the biggest power of two which is not subjected to 32-bit overflow (30 and not 31 to also avoid signed integer overflow)